pkg/manifests/tls_test.go

```diff
@@ -111,11 +112,36 @@ func TestRotateGrpcTLSSecret(t *testing.T) {
 			name: "force rotation",
 			setup: func(s *v1.Secret) {
 				s.Annotations["monitoring.openshift.io/grpc-tls-forced-rotate"] = "true"
+				// Introduce a delay to make sure that the initial and rotated
+				// certificates aren't created within the same second.
+				time.Sleep(1 * time.Second)
```

Comment:
I am guessing the below fails, can we not instead retry vs sleeping for one second? I guess since its unit tests its okay but still curious if we can approach a different way?

Reply:
I've added the delay because the new certificate might have exactly the same start/end as the original certificate so it isn't 100% guaranteed that `NotBefore` has actually been updated. I don't think that retrying would be easier to reason about.
I've tried to inject a custom `time.Now()` function in the tests but without success because we're mostly relying on `github.com/openshift/library-go/pkg/crypto` for which we can't replace `time.Now`.